src/osx/core/uilocale.mm

```diff
@@ -86,12 +88,48 @@ explicit wxUILocaleImplCF(NSLocale* nsloc)
         // Surprisingly, localeWithLocaleIdentifier: always succeeds, even for
         // completely invalid strings, so we need to check if the name is
         // actually in the list of the supported locales ourselves.
-        static wxCFRef<CFArrayRef>
-            all((CFArrayRef)[[NSLocale availableLocaleIdentifiers] retain]);
+//        static wxCFRef<CFArrayRef>
+//            all((CFArrayRef)[[NSLocale availableLocaleIdentifiers] retain]);
```

Comment:
This shouldn't remain commented out here, please feel free to cherry pick my commit if it's convenient or let me merge it into master already if you prefer -- just let me know if I should do it or not.

Reply:
I removed the commented out lines. Additionally, I replaced the `systemLocale` by the locale identified by **en_US_POSIX**. This will hopefully make the checks work again for macOS.

There is still some work to do regarding the handling of the standard C locale, that is, handling of special locale tags "C" and "POSIX".